lib/automation_client.js

```diff
@@ -557,40 +526,145 @@ AutomationApiClient.prototype.fetchTunnels = function(fnCallback) {
  * @param {Function} fnCallback is callback function.
  * @return {boolean} If tunnel is deleted successfully then return true otherwise false.
  */
-AutomationApiClient.prototype.deleteTunnelById = function(
-  tunnelId,
-  fnCallback
-) {
-  fnCallback = fnCallback || function() {};
-  if (typeof tunnelId === "function" || typeof tunnelId === "object") {
-    throw new Error("tunnelId is Required");
-  }
-  ApiClient.request(
-    {
-      url: "/tunnels/" + tunnelId,
-      method: "DELETE"
-    },
-    fnCallback
-  );
+AutomationApiClient.prototype.deleteTunnelById = function (tunnelId, fnCallback) {
+    fnCallback = fnCallback || function () {};
+    if (typeof tunnelId === "function" || typeof tunnelId === "object") {
+        throw new Error("tunnelId is Required");
+    }
+    ApiClient.request(
+        {
+            url: "/tunnels/" + tunnelId,
+            method: "DELETE",
+        },
+        fnCallback
+    );
 };
 
 /**
  * Add this method by Extending ApiClient for fetch Platforms.
  * @param {Function} fnCallback is callback function.
  * @return {!<Array>Platform|Error} return Platforms or Error.
  */
-AutomationApiClient.prototype.fetchPlatforms = function(fnCallback) {
-  fnCallback = fnCallback || function() {};
-  ApiClient.request(
-    {
-      url: "/platforms"
-    },
-    fnCallback
-  );
+AutomationApiClient.prototype.fetchPlatforms = function (fnCallback) {
+    fnCallback = fnCallback || function () {};
+    ApiClient.request(
+        {
+            url: "/platforms",
+        },
+        fnCallback
+    );
+};
+
+/**
+ * Add this method by Extending ApiClient for fetch sessions of latest build.
+ * @param {params} params is an object.
+ * @return {Object} return object of session listing.
+ */
+AutomationApiClient.prototype.getSessionsOfBuild = async function (params) {
+    // structure of response
+    // const data = {
+    //     success,
+    //     error,
+    //     msg
+    //     data,
+    // };
+
+    // fetching top builds sorted by descending order of their build ids
+    if (!!!params) {
+        return {
+            success: false,
+            error: "Build/Session params are required",
+            msg: "",
+            data: {},
+        };
+    }
+
+    // check for mandatory parameters
+    if (!!!params.buildName) {
+        return {
+            success: false,
+            error: "Build name is required",
+            msg: "",
+            data: {},
+        };
+    }
+
+    let buildParams = {
+        limit: params.buildLimit || 20,
+        sort: "desc.build_id",
+    };
+
+    let builds = {};
+    try {
+        builds = await this.fetchBuilds(buildParams);
+    } catch (error) {
+        console.error("Error fetching listing of builds --> ", error);
+        return {
+            success: false,
+            error: "Unable to fetch build listing",
+            msg: "",
+            data: {},
+        };
+    }
+
+    if (!!!builds.data || _.isEmpty(builds.data)) {
```

Comment:
Redundant !! , Can we replace !!! by ! ?

Reply:
Done